lib/codex/widgets/mods/mod_painter.dart

```diff
@@ -0,0 +1,217 @@
+import 'package:flutter/material.dart';
+import 'package:navis/codex/codex.dart';
+import 'package:warframestat_client/warframestat_client.dart';
+
+class CommonModPainter extends ModPainter {
+  const CommonModPainter({
+    required super.mod,
+    required super.rank,
+    required super.assets,
+  });
+
+  @override
+  double get effectsY => 375;
+
+  @override
+  double get ranksY => 440;
+}
+
+class LegendaryModPainter extends ModPainter {
+  const LegendaryModPainter({
+    required super.mod,
+    required super.rank,
+    required super.assets,
+  });
+
+  @override
+  double get effectsY => 380;
+
+  @override
+  double get ranksY => 444;
+}
+
+abstract class ModPainter extends CustomPainter {
+  const ModPainter({
+    required this.mod,
+    required this.rank,
+    required this.assets,
+  });
+
+  final Mod mod;
+  final int rank;
+  final ModAssets assets;
+
+  double get effectsY;
+  double get ranksY;
+
+  void drawEffects(Canvas canvas) {
+    final paint = Paint();
+    final frame = assets.frame;
+
+    canvas.drawImage(frame.cornerLights, Offset(195, effectsY), paint);
```

Comment:
this is gonna be annoying, but could you possibly document what each of these numbers corresponds to? just seems like a lot of magic numbers

Reply:
yeah I know but Dart has types for `Offset` so kinda already documented on what they do 
https://api.flutter.dev/flutter/dart-ui/Offset-class.html